README.md

```diff
@@ -1,7 +1,52 @@
-Schema-based data manipulation and code generation
+zcbor
+=====
+
+zcbor is a low footprint [CBOR](https://en.wikipedia.org/wiki/CBOR) library in the C language that comes with a schema-driven script tool that can validate your data, or even generate code for you.
+Aside from the script, the CBOR library is a standalone library which is tailored for use in microcontrollers.
+
+The validation part of the script can also work with YAML and JSON data.
+It can for example validate a YAML file against a schema and convert it into CBOR.
+
+The schema language used by zcbor is CDDL (Consise Data Definition Language) which is a powerful human-readable data description language defined in [IETF RFC 8610](https://datatracker.ietf.org/doc/rfc8610/).
+
+zcbor was previously called "cddl-gen".
+
+CBOR decoding/encoding library
+==============================
+
+The CBOR library found at [headers](include) and [source](src) is used by the generated code, but can also be used directly.
+If so, you must instantiate a `zcbor_state_t` object, which is most easily done using the `zcbor_new_state` function.
+
+The elem_count member refers to the number of encoded objects in the current list or map.
```

Comment:
Can we have variable names in their `inline` markdown form?

Reply:
I put elem_count in backticks.